viewer/main.cpp

```diff
@@ -182,38 +211,57 @@ int main(int argc, char **args)
             udSampleViewer_SaveSettings(settings);
           }
         }
-
-        ImGui::End();
       }
+      ImGui::End();
     }
     else
     {
       ImGui::SetNextWindowSize(ImVec2(0, 0));
-      if (ImGui::Begin("Samples"))
+      if (ImGui::BeginMainMenuBar())
       {
-        ImGui::LabelText("Samples", "Samples");
-        if (ImGui::BeginListBox("Samples"))
+        if (ImGui::BeginMenu("Samples"))
         {
-          for (size_t i = 0; i < udLengthOf(samples); ++i)
+          for (udSample *p : udSample::samples)
           {
-            if (ImGui::Selectable(udTempStr("%s##sample_%zu", samples[i].pName, i), selectedItem == i))
+            if (ImGui::Selectable(udTempStr("%s##sample_%p", p->GetName(), p), p == pCurrentSample))
             {
-              if (selectedItem >= 0)
+              if (pCurrentSample)
               {
-                samples[selectedItem].pDeinit(pSampleData);
-                pSampleData = nullptr;
+                pCurrentSample->Deinit();
+                pCurrentSample = nullptr;
               }
 
-              selectedItem = (int)i;
-
-              samples[selectedItem].pInit(&pSampleData, renderInfo);
+              pCurrentSample = p;
+              pCurrentSample->Init(renderInfo);
             }
           }
+          
+          isRunning = !ImGui::MenuItem("Exit", nullptr, nullptr);
 
-          ImGui::EndListBox();
+          ImGui::EndMenu();
+        }
+      }
+  
+      if (pCurrentSample)
+      {
+        bool exitRequested = pCurrentSample->Render(renderInfo);
```

Comment:
This doesn't return a `bool`, it returns a `udError`, perhaps returning a `bool` is all it needs to do.

Reply:
Yes that was the original implementation I had (returning bool) but I changed it when I made the base class so that future things could be more extensible. Will fix thanks